statistics-production/api-data-standards.qmd

```diff
@@ -0,0 +1,225 @@
+
+## Character limits for col_names and filter items
+
+Character limits for fields in data uploaded to the API are:
+
+::: {.table-responsive}
+
+| Element                         | Character limit |
+|---------------------------------|-----------------|
+|Filter / indicator column names  | 50 characters   |
+|Filter / indicator column labels | 80 characters   |
+|Filter items / location names    | 120 characters  |
+
+: Character limits on column names, column labels and filter items.
+
+:::
+
+## Tidy data structure
+
+The key thing on tidy data structure is to avoid filter items being included within indicator col_names. Where 
+you have collections of related terms appearing in indicator names (e.g. male, female, total), then these 
+should be translated into a filter column, with the data being pivoted.
+
+### Example 1 - Three metrics with a single filter
+
+#### Example of bad practice
+
+::: {.table-responsive}
+
+| school_count  | pupil_count_male | pupil_count_female | pupil_count_total | pupil_percent_male | pupil_percent_female | pupil_percent_total |
+|---------------|-------------------|---------------------|--------------------|---------------------|-----------------------|----------------------|
+| 2             | 120               | 130                 | 250               | 48                  | 52                    | 100                  |
+
+: Pupil counts and percentages in non-tidy format
+
+:::
+
+#### Example of good practice
+
+::: {.table-responsive}
+
+| sex                | school_count  | pupil_count  | pupil_percent  |
+|--------------------|---------------|--------------|----------------|
+| Male               | 2             | 30           | 60             |          
+| Female             | 2             | 40           | 80             |         
+| Total              | 2             | 50           | 100            |                  
+
+: Pupil counts and percentages in tidy format
+
+:::
+
+### Example 2 - Metrics with hierarchical filters
+
+#### Example of bad practice
+
+The following would not be accepted for publication via the API.
+
+::: {.table-responsive}
+
+| attendance |overall_absence | authorised_absence | unauthorised_absence | attendance_percent |overall_absence_percent | authorised_absence_percent | unauthorised_absence_percent |
+|------------|----------------|--------------------|----------------------|------------|----------------|-------------------|----------------------|
+| 180        | 20             | 12                 | 8                    | 90         | 10             | 6                 | 4                    | 
+
+: Attendance statistics in non-tidy format
+
+:::
+
+#### Example of good practice
+
+The following would be accepted for publication via the API. In this case, creating a hierarchical filter combination allows a clear representation of the data.
+
+::: {.table-responsive}
+
+| attendance_status  | attendance_type      | session_count  | session_percent |
```

Comment:
Mix of attendance and absence feels odd (example itself works well though), is this based on their real data or an example we've made up? 

(I ask as I wouldn't want a made up example to get questioned / bogged down in 'we'd never publish that anyway' style questions)

Reply:
Yep, literally straight from their dashboard data file... 
<img width="830" alt="image" src="https://github.com/user-attachments/assets/f88446e2-8111-4f7b-babc-f226e2f88bff">

Reply:
Fair enough! Happy with this